tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -981,6 +981,9 @@ class D3D12Device : public RendererBase
                 /// The type of binding in this range.
             slang::BindingType bindingType;
 
+                /// The shape of the resource
+            SlangResourceShape resourceShape;
```

Comment:
I'll be interested to see how this field is used/needed. I've gone to some trouble to try to keep the type related to ranges as small/lean as I can get away with, in order to avoid accidental bloat accumulating over time. Every single binding range will pay for this new field, even for non-resource ranges.

At the same time, this is probably a premature optimization concern on my part.